net/calendar-contacts-server/Portfile

```diff
@@ -0,0 +1,545 @@
+    reinplace \
+        "s|<string>${prefix}/var/run/calendarserver_proxy.pid</string>$|<string>${calendarserverdir}/Library/CalendarServer/run/nginx.pid</string>|" \
+        ${prefix}/etc/${startupitem.location}/org.macports.calendarserver_proxy/org.macports.calendarserver_proxy.plist
+
+    # TLS PFS
+    if ![file exists \
+        ${calendarserverdir}/Library/CalendarServer/etc/dhparam.pem] {
+        # create a shorter, faster DH parameter file for the default installation
+        system -W ${calendarserverdir}/Library/CalendarServer/etc \
+            "openssl dhparam -out dhparam.pem 2048"
+    }
+
+    fs-traverse f_or_d \
+        ${calendarserverdir} {
+        if { [file type ${f_or_d}] != "link" } {
+            file attributes ${f_or_d} \
+                 -owner ${calendarserverUser} -group _calendar
+        }
+    }
+}
+
+notes "Apple Calendar and Contacts Server is a standards-compliant
+server implementing the CalDAV and CardDAV protocols. Full
+deployment requires a working mail server, DNS configuration on both the
+LAN and the internet, including SPF and DKIM records, trusted TLS
+certificates, port forwarding, possibly a mail relay, and more.
+
+Users must reconfigure this installation for their own system, network,
+and security model specifics by editing all necessary files and checking
+file permissions. A subset of these settings are visible in the files:
+
+        port contents calendar-contacts-server
+        port file calendar-contacts-server
+
+These are the locations and network settings for the default configuration:
+
+    Calender and Contacts Server:
+        ${prefix}/var/calendarserver/Library/CalendarServer/Config/calendarserver.plist
+
+    nginx Reverse Proxy:
+        ${prefix}/var/calendarserver/Library/CalendarServer/etc/nginx.conf
+        ports: 8008, 8443, 8800, 8843
+
+    Personal data (note, outside ${prefix}):
+        /var/calendarserver/Library/CalendarServer/Data
+
+A working Calendar and Contacts Server will allow local account authentication
+at these web pages (ports 8008 and 8800 are unencrypted):
+
+        http://${fullhost}:8008
+        https://${fullhost}:8443
+        http://${fullhost}:8800
+        https://${fullhost}:8843
+
+Known issues:
+    * All local accounts have access to calendarserver's password using
+          security find-generic-password -a calendarserver@${fullhost} -g
+      This is a hack to provide twistedcaldav authentication access. See:
+      https://github.com/apple/ccs-calendarserver/blob/master/twistedcaldav/util.py
+      An alternate approach must be used if non-trusted local accounts
+      exist on the Calendar and Contacts Server.
+
+    * There is no destroot capability in ccs-calendarserver; all directories in
+        ${calendarserverdir}/Library/CalendarServer/${ccsname}
+      are not managed within `port {uninstall, contents, …} calendar-contacts-server`.
```

Comment:
It's of course normal that user data and config files are not registered to the port and are not altered or deleted on upgrade or uninstall. However, if the port is leaving executable code and related files behind, might a `post-deactivate` block be able to clean those up? We don't want to leave garbage behind on the user's system after they uninstall a port, and that goes for our automated build machines as well.

Reply:
Thanks, great suggestion! Done.